### PR TITLE
[Merged by Bors] - feat(RingTheory/IsTensorProduct): cancel pushout square on the left

### PR DESCRIPTION
**Pull request by chrisflav**
Given a commutative diagram of rings
```
  R  →  S  →  T
  ↓     ↓     ↓
  R' →  S' →  T'
```
where the left-hand square is a pushout and the big rectangle is a pushout, we show that then also the
right-hand square is a pushout.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/IsTensorProduct.lean [hunk 1]
```diff
@@ -471,4 +494,33 @@ theorem Algebra.IsPushout.algHom_ext [H : Algebra.IsPushout R S R' S'] {A : Type
...
+where the left-hand square is a pushout and the big rectangle is a pushout, then also the
+right-hand square is a pushout. Note that this is essentially the isomorphism
+`T ⊗[S] (S ⊗[R] R') ≃ₐ[T] T ⊗[R] R'`.
+-/
+lemma Algebra.IsPushout.of_comp {T' : Type*} [CommRing T'] [Algebra R T']
+    [Algebra S' T'] [Algebra S T'] [Algebra T T'] [Algebra R' T']
+    [IsScalarTower R T T'] [IsScalarTower S T T'] [IsScalarTower S S' T']
+    [IsScalarTower R R' T'] [IsScalarTower R S' T'] [IsScalarTower R' S' T']
+    [Algebra.IsPushout R S R' S'] [Algebra.IsPushout R T R' T'] :
+    Algebra.IsPushout S T S' T' := by
```
Comment (jcommelin) [inline Mathlib/RingTheory/IsTensorProduct.lean:513]: Can you please also add an iff version?

Comment (chrisflav) [reply Mathlib/RingTheory/IsTensorProduct.lean]: I replaced it with the `iff` version as this results in less code duplication. Should I provide the two directions as `comp` and `of_comp` separately or is this sufficient? (It certainly is for my application).

Comment (jcommelin) [reply Mathlib/RingTheory/IsTensorProduct.lean]: This is perfect

**Comment by jcommelin** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Comment by chrisflav** [comment]
Thanks for the reviews!

**Pull request closed without merging**